insights/parsers/sssd_conf.py

```diff
@@ -1,10 +1,86 @@
 """
-SSSD_Config - file ``/etc/sssd/sssd.config``
-============================================
+SSSD Configuration - files ``/etc/sssd/sssd.conf`` and ``/etc/sssd.conf.d/*``
+=============================================================================
+
+SSSD configuration files are in INI format.
+
+Example:
+    [sssd]
+    services = nss, sudo
+    domains = test
+
+    [sudo]
+    debug_level = 0xfff0
+
+    [domain/test]
+    id_provider = ldap
+    ldap_uri = ldap://ldap.test
+
+
+SSSDConf - files ``/etc/sssd/sssd.conf``
+----------------------------------------
+
+SSSDConfd - files under ``/etc/sssd/conf.d/*.conf``
+---------------------------------------------------
 """
+
 from insights.core import IniConfigFile
 from insights.core.plugins import parser
 from insights.specs import Specs
+from insights.util import deprecated
+
+
+@parser(Specs.sssd_config)
+class SSSDConf(IniConfigFile):
+    """
+    Parse the content of the ``/etc/sssd/sssd.conf``.
+
+    The file format is standard ini file.
+    """
+
+    def getboolean(self, section, option):
+        """
+        Returns:
+            bool: Returns boolean form based on the data from get.
+        """
+        val = self.get(section, option)
+        boolean_states = {
+            "true": True,
+            "false": False,
+        }
+
+        if val.lower() not in boolean_states:
+            raise ValueError("Not a boolean: %s" % val)
+
+        return boolean_states[val.lower()]
+
+    def set(self, section, option, value=None):
```

Comment:
I just noticed that this `set` method was added to the Parser (sorry for the late). Generally, this kind of `write-back-to-parser` operations is not recommended for insights. Since it's very common for insights to run a bunch of plugins serially in one pass of `insights-run`, if one of the plugins calls this `set` method and writes back to the Parser by mistake, it will affect the results of the following plugins that also depends on this Parser. (we have ever met such issue, and it's very difficult to debug :-().

Although, without this method, there are many ways can be used to write-back and pollution the parser indirectly,  here the `not recommended` means that we'd better do not provide such an interface publicly.

Reply:
I added a note and rename it to _set. I need this in the combiner to merge conf.d into the configuration. The parser is deepcopied in the container so the original parser is not changed. Let me know if you prefer different way.

Reply:
Looks good to me. Thanks for the change!